imcsdk/apis/admin/ntp.py

```diff
@@ -0,0 +1,175 @@
+                            Upto 4 ntp servers can be specified.
+
+    Returns:
+        CommNtpProvider object
+
+    Example:
+        ntp_enable(handle,
+                   ntp_servers = [{"id": 1, "ip": "192.168.1.1"},
+                                  {"id": 2, "ip": "192.168.1.2"}]
+    """
+
+    mo = _get_ntp_mo()
+    mo.ntp_enable = "yes"
+
+    _set_ntp_servers(mo, ntp_servers)
+    handle.set_mo(mo)
+    return handle.query_dn(mo.dn)
+
+
+def ntp_disable(handle):
+    """
+    Disable NTP
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        CommNtpProvider object
+    """
+
+    mo = _get_ntp_mo()
+    mo.ntp_enable = "no"
+
+    handle.set_mo(mo)
+    return handle.query_dn(mo.dn)
+
+
+def ntp_servers_modify(handle, ntp_servers=[]):
+    """
+    Modify the configured NTP servers
+    Args:
+        handle (ImcHandle)
+        ntp_servers (list): List of dictionaries of the type
+                            [{"id": 1, "ip": "192.168.1.1"},
+                             {"id": 2, "ip": "192.168.1.2"}]
+                            Upto 4 ntp servers can be specified.
+
+    Returns:
+        CommNtpProvider object
+
+    Example:
+        ntp_servers_modify(handle,
+                           ntp_servers = [{"id": 1, "ip": "192.168.1.1"},
+                                          {"id": 2, "ip": "192.168.1.2"},
+                                          {"id": 3, "ip": ""}]
+    """
+
+    # While sending the modified list of servers, it is imperative to send
+    # ntp_enable property in the request.
+    # Hence, query the MO and reassign the same value to ntp_enable
+    mo = _get_ntp_mo(handle, from_server=True)
+    mo.ntp_enable = mo.ntp_enable
+    _set_ntp_servers(mo, ntp_servers)
+
+    handle.set_mo(mo)
+    return mo
+
+
+def is_ntp_enabled(handle):
+    """
+    Check if NTP is enabled
+    Args:
+        handle (ImcHandle)
+
+    Returns:
+        bool
+    """
+
+    mo = _get_ntp_mo(handle, from_server=True)
+    return (mo.ntp_enable.lower() in ["true", "yes"])
+
+
+def _is_invalid_value(value):
+    return value in ["", None]
+
+
+def _check_ntp_server_match(ntp_mo, mo):
+    for prop in ["ntp_server1", "ntp_server2",
+                 "ntp_server3", "ntp_server4"]:
+        configured_value = getattr(ntp_mo, prop)
+        in_value = getattr(mo, prop)
+
+        if _is_invalid_value(configured_value) and \
+                _is_invalid_value(in_value):
+            continue
+        if configured_value != in_value:
+            return False
+
+    return True
+
+
+def ntp_setting_exists(handle, **kwargs):
+    """
+    Check if the specified NTP settings are already applied
+    Args:
+        handle (ImcHandle)
+        kwargs: key-value paired arguments
+
+    Returns:
+        (True, CommNtpProvider) if settings match, (False, None) otherwise
+    """
+
+    ntp_mo = _get_ntp_mo(handle, from_server=True)
+
+    if ntp_mo.ntp_enable != kwargs.get("ntp_enable"):
```

Comment:
Do we need to check if `ntp_enable` was passed in kwargs? get will give you an exception otherwise.. A user might just be checking if a entry exists, and not whether it is enabled or not.. Not sure, it makes a lot of sense to do that, but flexibility of checking only some things that are passed via kwargs is required.